test/src/test/java/hudson/PluginManagerTest.java

```diff
@@ -520,6 +519,25 @@ private void dynamicLoadAndDisable(String plugin) throws IOException, Interrupte
         assertNotNull(r.jenkins.getPluginManager().getPlugin("token-macro"));
     }
 
+    @Issue("JENKINS-36239")
+    @LocalData
+    @Test public void installPluginWithCycleDependency() {
+        List<String> pluginNames = new ArrayList<String>();
+        pluginNames.add("subversion");
+        try {
+            r.jenkins.pluginManager.install(pluginNames, true);
+        }
+        catch (StackOverflowError exception) {
+            fail("Detection of cyclic dependency failed");
+        }
+        catch (Failure exception) {
+            //should be thrown
```

Comment:
Is this scenario representative of users hitting JENKINS-36239? If so, does just changing the StackOverflowError to a CycleDetectedException actually help?